packages/core/admin/server/controllers/authenticated-user.js

```diff
@@ -34,6 +35,10 @@ module.exports = {
 
     const updatedUser = await userService.updateById(ctx.state.user.id, userInfo);
 
+    if (ctx.state.user.preferedLanguage !== updatedUser.preferedLanguage) {
+      sendDidChangeInterfaceLanguage();
```

Comment:
I don't know if already has been modified, but shouldn't we also send an event on admin creation?
Even if it's English by default, it'll update the payload sent (`['en']` => `['en', 'en']`)

Reply:
Also, is it wanted that we only send this event from the controller? What if a service update the `preferedLanguage` of some/all admin users using the user service?

Reply:
At first, I wanted to say in the `updateById` method of the admin user service, but it might be cleaner to write a [hook on `admin::user` update/create](https://docs.strapi.io/developer-docs/latest/development/backend-customization/models.html#lifecycle-hooks). What do you think?

Reply:
Also, don't forget about the [generic admin user update](https://github.com/strapi/strapi/blob/master/packages/core/admin/server/controllers/user.js#L70)

Reply:
Thanks a lot! I'm checking all of that